fsspec/spec.py

```diff
@@ -436,13 +436,21 @@ def glob(self, path, **kwargs):
         kwargs are passed to ``ls``.
         """
         import re
+        from glob import has_magic
 
         ends = path.endswith("/")
         path = self._strip_protocol(path)
-        indstar = path.find("*") if path.find("*") >= 0 else len(path)
-        indques = path.find("?") if path.find("?") >= 0 else len(path)
+        indstar = path.find("*")
+        indques = path.find("?")
+
+        if indstar < 0:
```

Comment:
I unwinding these necessary? Seems the change in the `if` below is the correction you were after.

Reply:
I unwound them though because in the master branch if `if path.find("*") >= 0` returned `True` then `path.find("*")` would be called a second time. This is an extremely small optimization though, It's not necessary, and if the previous style is prefered I'm happy to revert